alibi_detect/cd/base_online.py

```diff
@@ -86,6 +89,29 @@ def _configure_thresholds(self):
     def _configure_ref_subset(self):
         pass
 
+    @abstractmethod
+    def _update_state(self, x_t: Union[np.ndarray, list]):
+        pass
+
+    def _preprocess_xt(self, x_t: Union[np.ndarray, list]) -> np.ndarray:
```

Comment:
@ascillitoe I believe the `list` instead of `Any` type also manifests in other places such as here.

Reply:
Sorry yes you're right. The previous behavior was for `predict` to take `Any`, and then predict applied the preprocessing such that `score` was safe to accept `np.ndarray` only. With the new `_preprocess_xt` and `_update_state` all this needs a rethink. I shall look again at this now.

Reply:
Ok I think I've updated all the relevant `list`'s to `Any`. This includes `predict` in `cd/mmd_online.py` and `cd/lsdd_online.py`, which were currently `list` in the existing code, but I think should also be `Any`?